scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,349 @@
-    { x = -130, z = 458 },
-    { x = -118, z = 420 },
-    { x = -131, z = 381 },
-    { x = -212, z = 380 },
-    { x = -220, z = 392 },
-    { x = -219, z = 420 },
-    { x = -257, y = 12, z = 417 },
-    { x = -261, y = 8, z = 400 },
-    { x = -256, y = 4, z = 381 },
-    { x = -238, y = 0, z = 380 },
-    { x = -214, z = 381 },
-    { x = -209, z = 387 },
-    { x = -194, z = 388 },
-    { x = -179, z = 379 },
-    { x = -134, z = 379 },
-    { x = -115, z = 417 },
-    { x = -112, z = 454 },
-    { x = -105, z = 460 },
-    { x = -80, z = 460 },
-    { x = -65, z = 459 },
-    { x = -60, z = 452 },
-    { x = -59, z = 420 },
-    { x = 59, z = 420 },
-    { x = 60, z = 429 },
-    { x = 60, z = 445 },
-    { x = 62, z = 456 },
-    { x = 78, z = 460 },
-    { x = 108, z = 458 },
-    { x = 132, z = 381 },
-    { x = 185, z = 380 },
-    { x = 199, z = 391 },
-    { x = 218, z = 380 },
-    { x = 259, y = 4, z = 383 },
-    { x = 258, y = 12, z = 418 },
-    { x = 248, y = 14, z = 420 },
-    { x = 219, y = 16, z = 420 },
-    { x = 219, z = 386 },
-    { x = 210, z = 379 },
-    { x = 119, z = 380 },
-    { x = 140, z = 381 },
-    { x = 140, z = 419 },
-    { x = 175, z = 459 },
-    { x = 211, z = 459 },
-    { x = 220, z = 451 },
-    { x = 219, z = 420 },
-    { x = 248, y = 14, z = 420 },
-    { x = 258, y = 12, z = 418 },
-    { x = 259, y = 4, z = 383 },
-    { x = 218, y = 0, z = 380 },
-    { x = 199, z = 391 },
-    { x = 185, z = 380 },
-    { x = 132, z = 381 },
-    { x = 108, z = 458 },
-    { x = 78, z = 460 },
-    { x = 62, z = 456 },
-    { x = 60, z = 445 },
-    { x = 60, z = 429 },
-    { x = 59, z = 420 },
-    { x = -59, z = 420 },
-    { x = -60, z = 452 },
-    { x = -65, z = 459 },
-    { x = -80, z = 460 },
-    { x = -105, z = 460 },
-    { x = -112, z = 454 },
-    { x = -115, z = 417 },
-    { x = -134, z = 379 },
-    { x = -179, z = 379 },
-    { x = -191, z = 385 },
-    { x = -195, z = 396 },
-    { x = -202, z = 391 },
-    { x = -209, z = 387 },
-    { x = -214, z = 381 },
-    { x = -238, z = 380 },
-    { x = -256, y = 4, z = 381 },
-    { x = -261, y = 8, z = 400 },
-    { x = -257, y = 12, z = 417 },
-    { x = -219, y = 16, z = 420 },
-    { x = -220, z = 448 },
-    { x = -212, z = 459 },
-    { x = -130, z = 458 },
-    { x = -118, z = 420 },
-    { x = -131, z = 381 },
-    { x = -212, z = 380 },
-    { x = -220, z = 392 },
-    { x = -219, z = 420 },
-    { x = -257, y = 12, z = 417 },
-    { x = -261, y = 8, z = 400 },
-    { x = -256, y = 4, z = 381 },
-    { x = -238, y = 0, z = 380 },
-    { x = -214, z = 381 },
-    { x = -209, z = 387 },
-    { x = -202, z = 391 },
+    blueShort   = {x = -196, y = 0,  z = 389},
+    blueLong    = {x = -121, y = 16, z = 420},
+    yellowShort = {x =  191, y = 0,  z = 399},
+    yellowLong  = {x =  141, y = 16, z = 440},
 }
 
-entity.onMobSpawn = function(mob)
+-- Direction 0 = West -> East; 1 = East -> West
+local currentDirection = 0
```

Comment:
This can be named, like an enum;
```lua
local currentDirection =
{
    WEST_EAST = 0,
    EAST_WEST = 1, 
}
```

Reply:
If I change this to an enum, value assignment can longer be done on `currentDirection` directly without adding a new member.  Would something like this be okay to keep these values associated with each other?
```lua
local direction = 
{
    WEST_EAST = 0,
    EAST_WEST = 1,
    CURRENT   = 0,
}
```
Where `CURRENT` will track the numerical data, but `direction.WEST_EAST` and `direction.EAST_WEST` can be used for value assignment?

Reply:
I would go with defining
```lua
local direction = 
{
    WEST_EAST = 0,
    EAST_WEST = 1, 
}
```
And then just using the `currentDirection` variable to store it.

Another option to store that info is using an entity local variable (getLocalVar, setLocalVar)
But in any case, for now, I would just define the local enum and replacing the 0s and 1s with `direction.WEST_EAST` and `direction.EAST_WEST` and thats it

Reply:
Having said that, 0 and 1 are no better to read than EAST_WEST / WEST_EAST, IMO

Maybe `TO_WEST` and `TO_EAST` would be easier/better to read at a glance

Reply:
RIP - I did not refresh this page before amending again.
I ended up just adding `CURRENT` into the table/enum for now.
Being able to assign `direction.CURRENT = direction.EAST_WEST` and `direction.WEST_EAST` are definitely better than just 0/1.

I also opted to try and keep `CURRENT` in the `direction` table to keep that information centralized and not spread across multiple isolated variables.

Reply:
I can definitely change the values to `TO_EAST` and `TO_WEST` real quick though.